beginend.el

```diff
@@ -232,6 +232,32 @@ BEGIN-BODY and END-BODY are two `progn' expressions passed to respectively
   (progn
     (prodigy-last)))
 
+;; Copy/pasted from expand-region
+(defun beginend--point-is-in-comment-p ()
+  "Return non-nil if point is in comment."
+  (or (nth 4 (syntax-ppss))
+      (memq (get-text-property (point) 'face)
+            '(font-lock-comment-face font-lock-comment-delimiter-face))))
```

Comment:
The font-lock check is for the cases when you have the point on the delimiters, because there the syntax parser does not yet flag it as a comment (only inside the body, which is what the `syntax-ppss` tells you).  The extra checks in the code above handle the cases of being on the delimiters.

However, I suppose for our use-case here a more simplistic check is all that is needed as per my second comment.